contracts/Bridge/token/TokenController.sol

```diff
@@ -141,6 +145,37 @@ contract TokenController is Ownable, Pausable {
         accessControlManager = newAccessControlAddress_;
     }
 
+    /**
+     * @notice Migrates all minted tokens from one minter to another. This function is useful when we want to permanent take down a bridge.
+     * @param source_ Minter address to migrate tokens from.
+     * @param destination_ Minter address to migrate tokens to.
+     * @custom:access Controlled by AccessControlManager.
+     * @custom:error MintLimitExceed is thrown when the minting limit exceeds the cap after migration.
+     * @custom:event Emits MintLimitDecreased event for 'destination_'.
+     * @custom:event Emits MintedTokensMigrated.
+     */
+    function migrateMinterTokens(address source_, address destination_) external {
+        _ensureAllowed("migrateMinterTokens(address,address)");
+
+        uint256 sourceCap = minterToCap[source_];
+        uint256 destinationCap = minterToCap[destination_];
+
+        uint256 sourceMinted = minterToMintedAmount[source_];
+        uint256 destinationMinted = minterToMintedAmount[destination_];
+        uint256 newDestinationMinted = destinationMinted + sourceMinted;
+
+        if (newDestinationMinted > destinationCap) {
+            revert MintLimitExceed();
+        }
+
+        minterToMintedAmount[source_] = 0;
+        minterToMintedAmount[destination_] = newDestinationMinted;
+
+        uint256 availableLimit = destinationCap - newDestinationMinted;
+        emit MintLimitDecreased(destination_, availableLimit);
+        emit MintedTokensMigrated(source_, destination_);
```

Comment:
Should we emit `MintLimitIncreased(source_, sourceCap)`?

Reply:
This would be false alert because we won't allow any further minting from this specific minter.

Reply:
But that (disable the bridge) is not happening in this function. We are updating the storage variable, so I would emit the event. If we pause or reduce the mint cap later, a different event will be emitted and the full picture would make sense

Reply:
[8d388ac](https://github.com/VenusProtocol/token-bridge/pull/12/commits/8d388ac5dc9754e242bc7e9f43f45b51ea0d32d4)